overlord/state/taskrunner.go

```diff
@@ -482,6 +540,51 @@ ConsiderTasks:
 	return nil
 }
 
+// checkBlockignConditions removes the blockers that no longer apply and returns
+// true if none remain. Blockers are removed even if not all conditions were met,
+// so a condition only needs to be met once for it to be disregarded. If some
+// condition isn't met, returns the shortest of all check request periods if any
+// is set.
+// TODO: is it more useful to make it all or nothing? If one condition is not met,
+// we check all later w/o removing it?
+func (r *TaskRunner) checkBlockingConditions(t *Task) (bool, time.Duration, error) {
+	var removeIndexes []int
+	minCheckAfter := time.Duration(-1)
+	for i, blocker := range t.blockers {
+		ok, err := r.TestTaskBlocker(blocker)
+		if err != nil {
+			// maybe log and continue so we try again later?
+			return false, 0, err
+		}
+
+		if ok {
+			removeIndexes = append(removeIndexes, i)
+		} else {
+			if blocker.CheckAfter > 0 && (minCheckAfter == -1 || blocker.CheckAfter < minCheckAfter) {
+				minCheckAfter = blocker.CheckAfter
+			}
+		}
+	}
+
+	// remove blockers that no longer apply
+	if len(removeIndexes) == len(t.blockers) {
+		t.blockers = nil
+	} else {
+		for i, removeIndex := range removeIndexes {
+			// remove the blocker, accounting for the previously removed blockers
+			// when calculating the index
+			t.blockers = append(t.blockers[:removeIndex-i], t.blockers[removeIndex+1-i:]...)
```

Comment:
as blockers will be small wouldn't it be easier to build a new blockers list instead?